mne/morph.py

```diff
@@ -0,0 +1,1443 @@
+    verbose : bool | str | int | None
+        If not None, override default verbose level (see :func:`mne.verbose`
+        and :ref:`Logging documentation <tut_logging>` for more).
+
+
+    Attributes
+    ----------
+    kind : str | None
+        Kind of source estimate. E.g. 'volume' or 'surface'
+    subjects_dir : str | None
+        The path to the FreeSurfer subjects reconstructions.
+        It corresponds to FreeSurfer environment variable SUBJECTS_DIR.
+    subject_from : str | None
+        Name of the subject from which to morph as named in the SUBJECTS_DIR
+    subject_to : str | array | list of two arrays
+        Name of the subject on which to morph as named in the SUBJECTS_DIR
+        The default is 'fsaverage'. If morphing a surface source extimate,
+        subject_to can also be an array of vertices or a list of two arrays of
+        vertices to morph to. If morphing a volume source space, subject_to can
+        be the path to a MRI volume.
+    niter_affine : tuple of int
+        Number of levels (``len(niter_affine)``) and number of
+        iterations per level - for each successive stage of iterative
+        refinement - to perform the affine transform.
+        Default is niter_affine=(100, 100, 10)
+    niter_sdr : tuple of int
+        Number of levels (``len(niter_sdr)``) and number of
+        iterations per level - for each successive stage of iterative
+        refinement - to perform the Symmetric Diffeomorphic Registration (sdr)
+        transform. Default is niter_sdr=(5, 5, 3)
+    spacing : tuple | int | float | list | None
+        If morphing VolSourceEstimate, spacing is a tuple carrying the voxel
+        size of volume for each spatial dimension in mm.
+        If spacing is None, MRIs won't be resliced. Note that in this case
+        both volumes (used to compute the morph) must have the same number of
+        spatial dimensions.
+        If morphing SourceEstimate or Vector SourceEstimate, spacing can be
+        int, list (of two arrays), or None, defining the resolution of the
+        icosahedral mesh (typically 5). If None, all vertices will be used
+        (potentially filling the surface). If a list, then values will be
+        morphed to the set of vertices specified in in spacing[0] and
+        spacing[1].
+        Note that specifying the vertices (e.g., spacing=[np.arange(10242),
+        np.arange(10242)] for fsaverage on a standard spacing 5 source space)
+        can be substantially faster than computing vertex locations.
+    smooth : int | None
+        Number of iterations for the smoothing of the surface data.
+        If None, smooth is automatically defined to fill the surface
+        with non-zero values.
+    warn : bool
+        If True, warn if not all vertices were used.
+    xhemi : bool
+        Morph across hemisphere.
+    params : dict
+        The morph data. Contains all data relevant for morphing and / or volume
+        output.
```

Comment:
Since you have a `repr now`, you don't need all these attributes to be public, right? If you have public attributes, users may attempt to fiddle with them which can lead to unexpected results ... For the ones that you really need to expose, you can use `@property` for them